adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -453,16 +453,16 @@ private void updateOrderPaymentStatusAndInfo(OrderModel orderModel, PaymentsResp
     @Override
     public OrderData authorisePayment(final HttpServletRequest request, final CartData cartData) throws Exception {
         CheckoutCustomerStrategy checkoutCustomerStrategy = getCheckoutCustomerStrategy();
-        CustomerModel customer = null;
-        if (! checkoutCustomerStrategy.isAnonymousCheckout()) {
-            customer = checkoutCustomerStrategy.getCurrentUserForCheckout();
+
+        CustomerModel customer = checkoutCustomerStrategy.getCurrentUserForCheckout();
```

Comment:
This is changing logic here. If customer is guest customer and guestTokenization is not enabled, then too customer will be instantiated always. Is it intended? There could be some flows which are identifying guest customer by checking if(customer==null) condition.

Reply:
Talked offline.
Customer must be passed even on gust checkout, so shopper details can be passed on request.
Moved check for recurring contract flow for guest (isGuestUserTokenizationEnabled flag) to a later stage, on factory class.